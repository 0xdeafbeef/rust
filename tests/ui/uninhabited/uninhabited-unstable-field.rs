//@ aux-build: staged-api.rs
//@ revisions: current exhaustive
#![cfg_attr(exhaustive, feature(exhaustive_patterns))]
#![feature(never_type)]
#![feature(my_coro_state)] // Custom feature from `staged-api.rs`
#![deny(unreachable_patterns)]

extern crate staged_api;

use staged_api::{Foo, MyCoroutineState};

enum Void {}

fn demo(x: Foo<Void>) {
    match x {}
    //~^ ERROR non-exhaustive patterns
}

// Ensure that the pattern is not considered unreachable.
fn demo2(x: Foo<Void>) {
    match x {
        Foo { .. } => {}
    }
}

// Same as above, but for wildcard.
fn demo3(x: Foo<Void>) {
    match x {
        _ => {}
    }
}

fn unstable_enum(x: MyCoroutineState<i32, !>) {
    match x {
        //~^ ERROR non-exhaustive patterns
        MyCoroutineState::Yielded(_) => {}
    }
    match x {
        MyCoroutineState::Yielded(_) => {}
        MyCoroutineState::Complete(_) => {}
    }
}

fn main() {}
