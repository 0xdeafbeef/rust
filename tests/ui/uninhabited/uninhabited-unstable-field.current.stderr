error[E0004]: non-exhaustive patterns: type `Foo<Void>` is non-empty
  --> $DIR/uninhabited-unstable-field.rs:15:11
   |
LL |     match x {}
   |           ^
   |
note: `Foo<Void>` defined here
  --> $DIR/auxiliary/staged-api.rs:5:1
   |
LL | pub struct Foo<T> {
   | ^^^^^^^^^^^^^^^^^
   = note: the matched value is of type `Foo<Void>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern as shown
   |
LL ~     match x {
LL +         _ => todo!(),
LL ~     }
   |

error[E0004]: non-exhaustive patterns: `MyCoroutineState::Complete(_)` not covered
  --> $DIR/uninhabited-unstable-field.rs:34:11
   |
LL |     match x {
   |           ^ pattern `MyCoroutineState::Complete(_)` not covered
   |
note: `MyCoroutineState<i32, !>` defined here
  --> $DIR/auxiliary/staged-api.rs:11:1
   |
LL | pub enum MyCoroutineState<Y, R> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
LL |     Yielded(Y),
LL |     Complete(R),
   |     -------- not covered
   = note: the matched value is of type `MyCoroutineState<i32, !>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
   |
LL ~         MyCoroutineState::Yielded(_) => {},
LL +         MyCoroutineState::Complete(_) => todo!()
   |

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0004`.
